src/UI/Buyer/src/app/shared/services/favorites/favorites.service.ts

```diff
@@ -1,51 +1,43 @@
 import { Injectable, Inject } from '@angular/core';
-import {
-  OcMeService,
-  BuyerProduct,
-  Order,
-  MeUser,
-} from '@ordercloud/angular-sdk';
+import { OcMeService, BuyerProduct, Order } from '@ordercloud/angular-sdk';
 import { AppStateService } from '@app-buyer/shared';
 
 abstract class FavoritesService<T extends { ID?: string }> {
   protected readonly XpFieldName: string;
 
-  // Array of object IDs
-  public favorites: string[] = null;
-
   constructor(
     private appStateService: AppStateService,
     private ocMeService: OcMeService
-  ) { }
-
-  loadFavorites(): void {
-    if (this.favorites !== null) {
-      return;
-    }
-
-    this.appStateService.userSubject.subscribe((me: MeUser) => {
-      this.favorites =
-        me.xp && me.xp[this.XpFieldName] ? me.xp[this.XpFieldName] : [];
-    });
-  }
+  ) {}
 
   isFavorite(object: T): boolean {
-    return this.favorites !== null && this.favorites.indexOf(object.ID) > -1;
+    const favorites = this.getFavorites();
+    return favorites.includes(object.ID);
   }
 
   setFavoriteValue(isFav: boolean, object: T): void {
+    const favorites = this.getFavorites();
+    let updatedFavorites: string[];
     if (isFav) {
-      this.favorites.push(object.ID);
+      updatedFavorites = [...favorites, object.ID];
     } else {
-      this.favorites = this.favorites.filter((x) => x !== object.ID);
+      updatedFavorites = favorites.filter((x) => x !== object.ID);
     }
     const request = { xp: {} };
-    request.xp[this.XpFieldName] = this.favorites;
+    request.xp[this.XpFieldName] = updatedFavorites;
     this.ocMeService.Patch(request).subscribe((me) => {
       this.appStateService.userSubject.next(me);
-      this.favorites = me.xp[this.XpFieldName];
     });
   }
+
+  getFavorites(): string[] {
+    const me = this.appStateService.userSubject.value;
+    return me.xp &&
+      me.xp[this.XpFieldName] &&
```

Comment:
Also, you may not need `me.xp[this.XpFieldName]` at all since `instanceof Array` will return false for anything it would.

Reply:
good call, updated!